### ISSUE TITLE
#[repr] on a Rust enum with data in variants is counter-productive.

### ISSUE THREAD

**Issue by eddyb** [flier/rust-mqtt#4]
https://github.com/flier/rust-mqtt/blob/be90581515f4ce085c18299ad7b7fea03d047667/src/proto.rs#L25-L29

This actually means that *the tag* is `u8`, instead of being elided (because there's only one variant).

**Comment by flier** [comment]
I agree, I just want to use it like `Protocol::MQTT(4)`, maybe I can change it to a structure later

```rust
mod proto {
  pub struct MQTT(u8); 
}
```

**Comment by eddyb** [comment]
You can just remove the `#[repr(u8)]` attribute for now, it's a deoptimization basically.
